rust/arrow/src/record_batch.rs

```diff
@@ -36,7 +36,7 @@ use crate::error::{ArrowError, Result};
 /// serialization and computation functions, possibly incremental.  
 /// See also [CSV reader](crate::csv::Reader) and
 /// [JSON reader](crate::json::Reader).
-#[derive(Clone)]
+#[derive(Clone, Debug)]
```

Comment:
I don't see where you use the Debug

Reply:
I found it useful to have this trait for debugging purpose. It is not used in the change I made in this PR. I will remove it if the policy is to avoid adding Debug trait unless necessary.

Reply:
I'm curious what the debug out would be. Do you have an example?

Reply:
Here is what it looks like:

```
RecordBatch {
    schema: Schema {
        fields: [
            Field {
                name: "id",
                data_type: Int64,
                nullable: true,
                dict_id: 0,
                dict_is_ordered: false,
            },
        ],
        metadata: {},
    },
    columns: [
        PrimitiveArray<Int64>
        [
          5,
        ],
    ],
}
```

Again, I am happy to push a fixup to delete the Debug trait if anyone prefers that. I don't have a strong opinion on this, just thought it could be handy for future developers.

Reply:
I don't have a strong opinion in this either. If you found it useful for debugging then perhaps other users would too. @nevi-me What do you think? Any reason not to enable Debug here? Presumably there is no cost to this unless people choose to print them?

Reply:
I believe it does generate the code regardless even if it's not being used in debug print? if that's the case, it will increase size of binary.

Reply:
I think this is fine so went ahead and merged but we can revert the Debug change if anyone objects to this. Thanks for the fix @houqp !